internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -80,33 +81,41 @@ func (n *Networkd) Configure(ifaces ...*nic.NetworkInterface) error {
 		resolvers []net.IP
 	)
 
-	for _, iface := range ifaces {
-		// Bring up the interface
-		if err = n.Conn.LinkUp(&net.Interface{Index: int(iface.Index)}); err != nil {
-			log.Printf("failed to bring up %s: %v", iface.Name, err)
-			continue
-		}
+	var wg sync.WaitGroup
 
-		// Generate rtnetlink.AddressMessage for each address method defined on
-		// the interface
-		for _, method := range iface.AddressMethod {
-			log.Printf("configuring %s addressing for %s\n", method.Name(), iface.Name)
+	wg.Add(len(ifaces))
 
-			if err = n.configureInterface(method); err != nil {
-				// Treat as non fatal error when failing to configure an interface
-				log.Println(err)
-				continue
+	for _, iface := range ifaces {
+		go func(i *nic.NetworkInterface) {
```

Comment:
no `defer wg.Done()` ?

I bet this should hang forever...

Reply:
whoops

Reply:
Fixed